src/main/java/com/strandls/resource/services/Impl/ResourceServicesImpl.java

```diff
@@ -1005,4 +1022,85 @@ public MediaGalleryResourceData getResourceDataByID(Long rID) {
 		return null;
 	}
 
+	public static String generateFilePath(String directory, String fileName) {
+		int index = directory.indexOf("//");
+		if (index != -1) {
+			directory = "/" + directory.substring(index + 2);
+		}
+		String filepath = directory + '/' + fileName;
+		return filepath;
+	}
+
+	@Override
+	public Response getImage(HttpServletRequest request, Long resourceId, Integer width, Integer height, String format,
+			String fit, boolean preserve) {
+		try {
+
+			Resource resource = resourceDao.findById(resourceId);
+			if (resource == null) {
+				return Response.status(Status.NOT_FOUND).entity("Resource not found").build();
+			}
+			if (!"IMAGE".equals(resource.getType())) {
+				return Response.status(Status.NOT_FOUND).entity("Resource dont have an image").build();
+			}
+
+			String prefix = ResourceUtil.folderPrefix(resource.getContext());
+			Path path = Paths.get(prefix + resource.getFileName());
+			String directory = path.getParent().toString();
+			String fileName = path.getFileName().toString();
+			String dirPath = storageBasePath + File.separatorChar + directory + File.separatorChar;
+
+			String fileLocation = dirPath + fileName;
+			File file = ResourceUtil.findFile(fileLocation);
+
+			UserIbp userIbp = userService.getUserIbp(resource.getUploaderId().toString());
+			String watermark = "";
+			if (Boolean.TRUE.equals(Boolean.parseBoolean(waterMarked))) {
+				watermark = resource.getContributor() != null ? resource.getContributor() : userIbp.getName();
+			}
+
+			if (file == null) {
+				return Response.status(Status.NOT_FOUND).entity("File not found").build();
+			}
+
+			String name = file.getName();
+			String extension = name.substring(name.indexOf(".") + 1);
+			String thumbnailFolder = storageBasePath + File.separatorChar + BASE_FOLDERS.THUMBNAILS.getFolder()
+					+ file.getParentFile().getAbsolutePath().substring(storageBasePath.length());
+
+			String command = null;
+			command = ResourceUtil.generateImageCommand(file.getAbsolutePath(), thumbnailFolder, width, height,
+					preserve ? extension : format, Integer.parseInt(imageQuality), fit, watermark);
+
+			File resizedFile = getResizedFile(command, thumbnailFolder, file, Integer.parseInt(imageQuality));
+			Tika tika = new Tika();
+			String detactedContentType = tika.detect(resizedFile.getName());
+			String contentType = ResourceUtil.determineContentType(preserve, format, detactedContentType);
+
+			return ResourceUtil.fromFileToStream(resizedFile, contentType);
+		} catch (FileNotFoundException fe) {
+			logger.error(fe.getMessage());
+			return Response.status(Status.NOT_FOUND).build();
+		} catch (Exception ex) {
+			logger.error(ex.getMessage());
+			return Response.status(Status.INTERNAL_SERVER_ERROR).build();
+		}
+	}
+
+	private File getResizedFile(String command, String thumbnailFolder, File file, int imageQuality) {
```

Comment:
I have few confusions here in the flow. Please get in touch so that I am more aware of the flow.